ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -258,16 +258,16 @@ def collect(self, device, unused):
     def process(self, device, results, unused):
         tenants = []
         for tenant in results['tenants']:
-            if tenant['enabled'] is not True:
+            if tenant.get('enabled', False) is not True:
                 continue
 
             tenants.append(ObjectMap(
                 modname='ZenPacks.zenoss.OpenStackInfrastructure.Tenant',
                 data=dict(
-                    id=prepId('tenant-{0}'.format(tenant['id'])),
-                    title=tenant['name'],               # nova
-                    description=tenant['description'],  # tenant description
-                    tenantId=tenant['id']
+                    id=prepId('tenant-{0}'.format(tenant.get('id', ''))),
```

Comment:
I think for all of them, if we can't determine an ID, we should skip the objectmap altogether- there's really no point in trying to load a tenant with an ID of 'tenant-'.  It's just going to fail anyway if there's more than one.